pym/portage/dbapi/vartree.py

```diff
@@ -2491,7 +2498,10 @@ def unlink(file_name, lstatobj):
 								(statobj.st_dev, statobj.st_ino),
 								[]).append(relative_path)
 
-					if is_owned:
+					# if the file was replaced by new version, keep it
+					# unless it is also INSTALL_MASK-ed in the new version
+					# (then it was not really installed and we want to clean it)
+					if is_owned and not self._is_install_masked(relative_path[1:], install_mask):
```

Comment:
Since contents contains files matches by INSTALL_MASK, we should also not unmerge files that were matched by INSTALL_MASK at install time for the package that is currently being unmerged.

Reply:
Wrong. Then you will be not remove files that were installed previously but now are masked.

Reply:
Wow this is confusing. Suppose that nothing is masked now. We don't want to unmerge files that were masked at install time for the package that is currently being unmerged, because those files don't really belong to this package.

Reply:
Maybe it's a big mistake to record masked files in CONTENTS. Why are we doing that? If we need that info, maybe a separate file from CONTENTS is more appropriate?